src/ring_theory/ideal/operations.lean

```diff
@@ -1128,6 +1128,13 @@ le_antisymm (map_le_iff_le_comap.2 $ mul_le.2 $ λ r hri s hsj,
   by rw [← f.map_mul];
   exact mem_map_of_mem f (mul_mem_mul hri hsj))
 
+theorem map_pow (n : ℕ) : map f (I^n) = (map f I)^n :=
```

Comment:
@eric-wieser Should there be some `ideal.map_mul_hom` that golfs this proof? Where should it go?

Reply:
Sure! Should I remove `map_pow` from the PR?

Reply:
The lemma `comap_le_comap_iff_of_surjective` also feels a bit out of pace with the direction this PR has taken.

Reply:
if you need these lemmas, just leave them in! the follow-up comment was about making an `ideal.map_mul_hom`.